### ISSUE TITLE
Select-less version of {Integer,Long}.highestOneBit

### ISSUE THREAD

**Issue by sjrd** [scala-native/scala-native#256]
I found a branchless and select-less version of `{Integer,Long}.highestOneBit`:
This is the current version for `Integer`:
```scala
  @inline def highestOneBit(i: scala.Int): scala.Int =
    select(i == 0, 0, (1 << 31) >>> numberOfLeadingZeros(i))
```
and it can be instead:
```scala
  @inline def highestOneBit(i: scala.Int): scala.Int = {
    /* The natural way of implementing this is:
     *   if (i == 0) 0
     *   else (1 << 31) >>> numberOfLeadingZeros(i)
     *
     * We can deal with the 0 case in a branchless fashion by adding `& i` to
     * the else branch:
     *   ((1 << 31) >>> numberOfLeadingZeros(i)) & i
     * Indeed, when i == 0, the `& i` collapses everything to 0. And otherwise,
     * we know that ((1 << 31) >>> numberOfLeadingZeros(i)) is the highest 1
     * bit of i, so &'ing with i is a no-op.
     *
     * Finally, since we're &'ing with i anyway, we can replace the >>> by a
     * >>, which is shorter in JS and does not require the additional `| 0`.
     */
    ((1 << 31) >>> numberOfLeadingZeros(i)) & i
  }
```
with the same code for `Long`.

See https://github.com/scala-js/scala-js/pull/2546/files#diff-d559f7f168fcc181b40dc5c47cb7aa8b

**Comment by densh** [comment]
Thanks @sjrd, we should try totally try this out.

**Comment by densh** [comment]
Fixed in #264
